apps/files/utils.py

```diff
@@ -91,49 +91,77 @@ class Extractor(object):
     def parse(cls, path):
         install_rdf = os.path.join(path, 'install.rdf')
         package_json = os.path.join(path, 'package.json')
+        manifest_json = os.path.join(path, 'manifest.json')
         if os.path.exists(install_rdf):
             return RDFExtractor(path).data
         elif os.path.exists(package_json):
             return PackageJSONExtractor(package_json).parse()
+        elif os.path.exists(manifest_json):
+            return ManifestJSONExtractor(manifest_json).parse()
         else:
-            raise forms.ValidationError("No install.rdf or package.json found")
+            raise forms.ValidationError(
+                "No install.rdf or package.json or manifest.json found")
 
 
-class PackageJSONExtractor(object):
-    def __init__(self, path):
+def get_appversions(app, min_version, max_version):
+    """Return the `AppVersion`s that correspond to the given versions."""
+    qs = AppVersion.objects.filter(application=app.id)
+    min_appver = qs.get(version=min_version)
+    max_appver = qs.get(version=max_version)
+    return min_appver, max_appver
+
+
+def get_simple_version(version_string):
```

Comment:
I'm not too keen on using external dependencies for such a simple and basic task. Also, we don't want something too clever, which would extract version number, or worse, fail on badly formed version numbers.